pingcap/tidb
# session: fix the bug that may cause upgrading from v4.0 fail

### What problem does this PR solve?

Issue Number: partially handles #21321

Problem Summary:


When cherry-picking upgradeToVer52 to v4.0, we wrongly name it upgradeToVer48.
When cherry-picking upgradeToVer54 to v4.0, we wrongly name it upgradeToVer49.
If we upgrade from v4.0 to a newer version, the real upgradeToVer48 and upgradeToVer49 will be missed.


So we redo upgradeToVer48 and upgradeToVer49 in upgradeToVer55 and upgradeToVer56 to make sure the upgrading from v4.0 succeeds.
Unneeded versions(version39 in #17293, version48 and version49) are also deleted.

### Check List 

Tests 

- Unit test
- Integration test
- Manual test (add detailed scripts or steps below)
git checkout release-4.0
make && run
exit
git checkout this_pr
make && run
check if it succeeds and `mysql.stats_extended` exists and `tidb_index_lookup_concurrency` is set tot -1


### Release note 

- no release note
